src/encodings.jl

```diff
@@ -1,271 +1,384 @@
-                  "shift_jis", "SHIFT_JIS", "SHIFT_JISX0213", "shift-jis", "SHIFT-JIS",
-                  "SJIS", "SJIS-MS", "SJIS-OPEN", "SJIS-WIN", "SS636127", "ST_SEV_358-88",
-                  "STRK1048-2002", "T.61", "T.61-8BIT", "T.618BIT", "TCVN", "TCVN-5712",
-                  "TCVN5712-1", "TCVN5712-1:1993", "THAI8", "TIS-620", "TIS620",
-                  "TIS620-0", "TIS620.2529-1", "TIS620.2533-0", "TIS620.2533-1",
-                  "TS-5881", "TSCII", "TURKISH8", "UCS-2", "UCS-2-INTERNAL", "UCS-2-SWAPPED",
-                  "UCS-2BE", "UCS-2LE", "UCS-4", "UCS-4-INTERNAL", "UCS-4-SWAPPED",
-                  "UCS-4BE", "UCS-4LE", "UCS2", "UCS2BE", "UCS2LE", "UCS4", "UCS4BE",
-                  "UCS4LE", "UHC", "UJIS", "UK", "UNICODE", "UNICODE-1-1", "UNICODE-1-1-UTF-7",
-                  "UNICODEBIG", "unicodeFFFE", "UNICODELITTLE", "US", "us-ascii",
-                  "US-ASCII", "UTF-16", "UTF-16BE", "UTF-16LE", "UTF-32", "UTF-32BE",
-                  "UTF-32LE", "UTF-7", "UTF-8", "UTF16", "UTF16BE", "UTF16LE",
-                  "UTF32", "UTF32BE", "UTF32LE", "UTF7", "UTF8", "VISCII", "VISCII1.1-1",
-                  "WCHAR_T", "WIN-SAMI-2", "WINBALTRIM", "windows-1250", "WINDOWS-1250",
-                  "windows-1251", "WINDOWS-1251", "windows-1252", "WINDOWS-1252",
-                  "windows-1253", "WINDOWS-1253", "windows-1254", "WINDOWS-1254",
-                  "windows-1255", "WINDOWS-1255", "windows-1256", "WINDOWS-1256",
-                  "windows-1257", "WINDOWS-1257", "windows-1258", "WINDOWS-1258",
-                  "WINDOWS-31J", "WINDOWS-50221", "WINDOWS-51932", "windows-874",
-                  "WINDOWS-874", "WINDOWS-932", "WINDOWS-936", "WINSAMI2", "WS2",
-                  "x_Chinese-Eten", "x-Chinese_CNS", "x-cp20001", "x-cp20003",
-                  "x-cp20004", "x-cp20005", "x-cp20261", "x-cp20269", "x-cp20936",
-                  "x-cp20949", "x-cp50227", "x-EBCDIC-KoreanExtended", "x-Europa",
-                  "x-IA5", "x-IA5-German", "x-IA5-Norwegian", "x-IA5-Swedish",
-                  "x-iscii-as", "x-iscii-be", "x-iscii-de", "x-iscii-gu", "x-iscii-ka",
-                  "x-iscii-ma", "x-iscii-or", "x-iscii-pa", "x-iscii-ta", "x-iscii-te",
-                  "x-mac-arabic", "x-mac-ce", "x-mac-chinesesimp", "x-mac-chinesetrad",
-                  "x-mac-croatian", "x-mac-cyrillic", "x-mac-greek", "x-mac-hebrew",
-                  "x-mac-icelandic", "x-mac-japanese", "x-mac-korean", "x-mac-romanian",
-                  "x-mac-thai", "x-mac-turkish", "x-mac-ukrainian", "X0201", "X0208",
-                  "X0212", "YU"]
+# This file is a part of StringEncodings.jl. License is MIT: http://julialang.org/license
+
+# Parametric singleton type representing a given string encoding via its symbol parameter
+
+import Base: show, print, convert
+export Encoding, @enc_str, codeunit, native_endian
+
+immutable Encoding{enc} end
```

Comment:
Being able to statically compile things like this will be very important to the sort of work we're doing, we want to be able to start up julia sessions to run pre compiled apps in containers as quick as possible.  If using generated functions makes that not possible, then I'd really like to see if it can be accomplished in a different manner.